packages/repository/src/model.ts

```diff
@@ -145,6 +149,63 @@ export class ModelDefinition {
     return this;
   }
 
+  /**
+   * Define a new "belongs to" relation.
+   * @param name - The name of the "belongs to" relation.
+   * @param definition - The definition for the "belongs to" relation.
+   */
+  belongsTo(
+    name: string,
+    definition: Partial<BelongsToDefinition> &
+      Pick<BelongsToDefinition, 'source' | 'target'>,
+  ): this {
+    const belongsToDefinition: BelongsToDefinition = {
+      ...definition,
+      name,
+      type: RelationType.belongsTo,
+      targetsMany: false,
```

Comment:
Is it possible to infer `keyFrom` value from the target model name? Let's say `target` is `() => Customer` and we are defining the relation "order belongs to a customer", would something like this work?

```ts
if (! belongsToDefinition.keyFrom) {
  const targetName = definition.target().modelName;
  // ^^ should be "Customer"
  belongsToDefinition.keyFrom = camelCase(`${targetName}_Id`);
  // ^^ should produce "customerId"
}
```

A similar comment applies to other relation types too - `hasOne` and `hasMany` can use `definition.source.modelName` to build the foreign key name.

Since this is an incremental addition, I am ok to leave it out of this initial version if you prefer.

Reply:
Oh, I see that you are computing `keyFrom` in `resolveBelongsToMetadata`, that's a good solution too 👍

Reply:
I tried your suggestion and adding the logic inside the belongsTo function may lead to reference errors if they define the target class after the class that is using the belongsTo method, so adding the logic in `resolveBelongsToMetadata` is more flexible.

e.g.,
https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/__tests__/integration/repositories/relation.factory.integration.ts#L464

The `Customer` class variable is available in the scope, but has not been initialized yet so it will throw a reference error if we try to access it.